tests/native/test_armv7cpu.py

```diff
@@ -13,25 +13,266 @@
+        "sbc r3, r1, #24, 20": b"183ac1e2",
+        "stm sp, {r1, r2, r3}": b"0e008de8",
+        "stm sp!, {r1, r2, r3}": b"0e00ade8",
+        "stmda r0!, {r1, r2, r3}": b"0e0020e8",
+        "stmdb r0!, {r1, r2, r3}": b"0e0020e9",
+        "stmia r0!, {r1, r2, r3}": b"0e00a0e8",
+        "stmib r0!, {r1, r2, r3}": b"0e00a0e9",
+        "str R2, [R1]": b"002081e5",
+        "str SP, [R1]": b"00d081e5",
+        "str R1, [R2, R3]": b"031082e7",
+        "str R1, [R2, R3, LSL #3]": b"831182e7",
+        "str R1, [R2, #3]!": b"0310a2e5",
+        "str R1, [R2], #3": b"031082e4",
+        "strd R2, [R1]": b"f020c1e1",
+        "sub r3, r1, r2": b"023041e0",
+        "sub r3, r1, #5": b"053041e2",
+        "sub r3, r1, #0x18000": b"063941e2",
+        "sub r3, r1, #24, 20": b"183a41e2",
+        "svc #0": b"000000ef",
+        "sxth r1, r2": b"7210bfe6",
+        "sxth r3, r4": b"7430bfe6",
+        "sxth r5, r4, ROR #8": b"7454bfe6",
+        "teq r3, r1": b"010033e1",
+        "teq r3, #0x18000": b"060933e3",
+        "teq r3, #24, 20": b"180a33e3",
+        "BIC R1, #0x10": b"1010c1e3",
+        "tst r3, #0x18000": b"060913e3",
+        "tst r3, #24, 20": b"180a13e3",
+        "UMULLS R1, R2, R1, R2": b"911292e0",
+        "uqsub8 r3, r1, r2": b"f23f61e6",
+        "uxtb r1, r2": b"7210efe6",
+        "uxth r1, r2": b"7210ffe6",
+        "vldmia  r1, {d8, d9, d10}": b"068b91ec",
+        "vldmia  r1!, {d8, d9, d10}": b"068bb1ec",
+    },
+    CS_MODE_THUMB: {
+        "adds r0, #4": b"0430",
+        "addw r0, r1, #0x2a": b"01f22a00",
+        "addw r0, pc, #0x2a": b"0ff22a00",
+        "adr r0, #16": b"04a0",
+        "asr.w R5, R6, #3": b"4feae605",
+        "cbnz r0, #0x2a": b"98b9",
+        "cbz r0, #0x2a": b"98b1",
+        "cmp r1, #1": b"0129",
+        "ite ne": b"14bf",
+        "mov r2, r12": b"6246",
+        "mov r3, r12": b"6346",
+        "mov r4, r12": b"6446",
+        "itete ne": b"15bf",
+        "mov r1, #1": b"4ff00101",
+        "mov r2, #1": b"4ff00102",
+        "mov r3, #1": b"4ff00103",
+        "mov r4, #4": b"4ff00404",
+        "itt ne": b"1cbf",
+        "lsl.w r5, r6, #3": b"4feac605",
+        "lsr.w R5, R6, #3": b"4fead605",
+        "lsr.w R0, R0, R2": b"20fa02f0",
+        "orn r2, r2, r5": b"62ea0502",
+        "sbcs r0, r3": b"9841",
+        "sel r4, r5, r6": b"a5fa86f4",
+        "subw r0, r1, #0x2a": b"a1f22a00",
+        "subw r0, pc, #0x2a": b"aff22a00",
+        "  tst r0, r0\n  beq label\n  bne label\nlabel:\n  nop": b"004200d0ffd100bf",
+        "tbb [r0, r1]": b"d0e801f0",
+        "tbb [pc, r1]": b"dfe801f0",
+        "tbh [r0, r1, lsl #1]": b"d0e811f0",
+        "tbh [pc, r1, lsl #1]": b"dfe811f0",
+        "adcs r3, r4": b"6341",
+        "eor r3, #5": b"83f00503",
+        "lsrs r1, r2": b"d140",
+        "orr r3, #5": b"43f00503",
+        "sub r3, #12": b"a3f10c03",
+        "uadd8 r2, r2, r3": b"82fa43f2",
+    },
+}
+
+
+def _ks_assemble(asm, mode=CS_MODE_ARM):
```

Comment:
```suggestion
def _ks_assemble(asm: str, mode=CS_MODE_ARM) -> bytes:
    """Assemble the given string using Keystone using the specified CPU mode."""
    # Explicitly uses late importing so that Keystone will only be imported if this is called.
    # This lets us avoid requiring installation of Keystone for running tests.
```